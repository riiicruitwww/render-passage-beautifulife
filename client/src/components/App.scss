.App {
  height: 100vh;

  &__body {
    display: flex;
    align-items: center;
    justify-content: center;
    height: calc(100vh - 70px);
  }

  @media (max-width: 700px) {
    &__body {
      flex-direction: column;
    }
  }
}
