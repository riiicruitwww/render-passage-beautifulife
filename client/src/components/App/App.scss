.App {
  display: flex;
  flex-direction: column;
  height: 100vh;

  &__body {
    flex: 1;
    display: flex;
    align-items: flex-start;
    justify-content: center;
    padding-top: 1.5em;
    background-color: rgb(16, 16, 16);
  }

  @media (max-width: 700px) {
    &__body {
      flex-direction: column;
      padding-top: 0px;
    }
  }
}
