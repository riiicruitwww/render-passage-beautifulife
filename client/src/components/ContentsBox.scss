.ContentsBox {
  flex: 1;
  overflow: auto;
  max-width: 600px;
  max-height: calc(100vh - 110px - 6em);
  margin: 1.5em;
  padding: 1em 1.5em;
  padding-bottom: 0;
  border: 1px solid black;
  border-radius: 5px;
  background-color: rgb(255, 255, 255);
}
