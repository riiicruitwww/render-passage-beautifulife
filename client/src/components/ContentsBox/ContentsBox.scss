.ContentsBox {
  flex: 1;
  overflow: auto;
  max-width: 600px;
  max-height: calc(100vh - 206px);
  margin: 24px;
  padding: 16px 24px 0 24px;
  border: 1px solid #000000;
  border-radius: 5px;
  background-color: #ffffff;
}
