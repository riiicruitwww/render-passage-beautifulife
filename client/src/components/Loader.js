import React from 'react';
import './Loader.scss';

const Loader = () => {
  return (
    <div className="Loader">
      <div className="Loader__circle1" />
      <div className="Loader__circle2" />
    </div>
  );
};

export default Loader;
