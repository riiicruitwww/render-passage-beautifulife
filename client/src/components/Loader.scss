.Loader {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.1);

  div {
    position: absolute;
    top: calc(50% - 80px);
    left: calc(50% - 80px);
    padding: 80px;
    border-radius: 50%;
  }

  &__circle1 {
    background-color: rgb(239, 94, 94);
    animation: circle1 1.5s ease-out infinite alternate-reverse;
  }

  &__circle2 {
    background-color: rgb(81, 91, 229);
    animation: circle1 1.5s ease-out infinite alternate-reverse;
  }
}

@keyframes circle1 {
  0% {
    transform: scale(0);
    opacity: 1;
  }
  100% {
    transform: scale(1);
    opacity: 0;
  }
}
