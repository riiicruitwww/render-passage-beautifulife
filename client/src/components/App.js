import React, { Component, Fragment } from 'react';
import './App.scss';
import Header from './Header';
import ContentsBox from './ContentsBox';
import { renderPropertyByName } from './renderPropertyByName';

class App extends Component {
  componentDidMount() {
    const { onInit } = this.props;

    onInit();
  }

  renderPassgeBox() {
    const { passageBox } = this.props;

    const renderedPassageBox = renderPropertyByName(passageBox.view_tree);
    const renderedPassages = passageBox.passages.map((passage, index) => {
      return renderPropertyByName(passage.view_tree, index);
    });

    return (
      <Fragment>
        {renderedPassageBox}
        {renderedPassages}
      </Fragment>
    );
  }

  renderQuestions() {
    const { questions } = this.props;

    return questions.map((question, index) => {
      return (
        <div key={index} className="question-wrapper">
          <span className="order">{question.order + 1}.</span>
          {renderPropertyByName(question.view_tree, index)}
        </div>
      );
    });
  }

  render() {
    const { packType } = this.props;

    return (
      <div className="App">
        <Header />
        {packType && (
          <div className="App__body">
            <ContentsBox>{this.renderPassgeBox()}</ContentsBox>
            <ContentsBox>{this.renderQuestions()}</ContentsBox>
          </div>
        )}
      </div>
    );
  }
}

export default App;
