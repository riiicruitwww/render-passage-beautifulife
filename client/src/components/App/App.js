import React, { Component, Fragment } from 'react';
import './App.scss';
import { Divider, Loader } from 'Base';
import {
  Paragraph,
  Passage,
  Vocabulary
} from 'Contents'
import {ContentsBox, Header } from 'Template';
import QuestionContainer from 'containers/QuestionContainer';

class App extends Component {
  componentDidMount() {
    const { location, onInit } = this.props;

    onInit(location.pathname);
  }

  renderPassageBox() {
    const { passageBox: { passages, view_tree } } = this.props;

    const renderPassageBoxParagraph = () => {
      return view_tree.children.map((item, index) => (
        <Paragraph key={index} paragraph={item} />
      ));
    };

    const renderPassages = () => {
      return passages.map((passage, index) => (
        <Passage key={index} passage={passage} />
      ));
    };

    return (
      <Fragment>
        {renderPassageBoxParagraph()}
        {renderPassages()}
      </Fragment>
    );
  }

  renderPassageTranslations() {
    const { passageBox } = this.props;

    return passageBox.passage_translations.map((passage, index) => (
      <Passage key={index} passage={passage} />
    ));
  }

  renderVocabularies() {
    const { passageBox } = this.props;

    return passageBox.vocabularies.map((vocabulary, index) => (
      <Vocabulary key={index} vocabulary={vocabulary} />
    ));
  }

  renderQuestions() {
    const { questions } = this.props;

    return questions.map((question, index) => (
      <QuestionContainer key={index} question={question} />
    ));
  }

  render() {
    const {
      isAnswerComplete,
      isCheckComplete,
      isLoading,
      packType,
      onCheckClick,
      onNextClick,
      onHomeClick
    } = this.props;

    return (
      <div className="App">
        <Header
          isAnswerComplete={isAnswerComplete}
          isCheckComplete={isCheckComplete}
          packType={packType}
          onCheckClick={onCheckClick}
          onNextClick={onNextClick}
          onHomeClick={onHomeClick}
        />
        <div className="App__body">
          {packType && (
            <Fragment>
              <ContentsBox>{this.renderPassageBox()}</ContentsBox>
              <ContentsBox>
                <p>&#8226;&nbsp;&nbsp;Questions</p>
                {this.renderQuestions()}
                {isCheckComplete && (
                  <Fragment>
                    <Divider />
                    <p>&#8226;&nbsp;&nbsp;해석</p>
                    {this.renderPassageTranslations()}
                    <Divider />
                    <p>&#8226;&nbsp;&nbsp;단어</p>
                    {this.renderVocabularies()}
                  </Fragment>
                )}
              </ContentsBox>
            </Fragment>
          )}
        </div>
        {isLoading && <Loader />}
      </div>
    );
  }
}

export default App;
