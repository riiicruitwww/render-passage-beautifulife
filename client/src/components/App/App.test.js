import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import Loader from '../Loader/Loader';
import ContentsBox from '../ContentsBox/ContentsBox';

App.prototype.renderPassgeBox = jest.fn();
App.prototype.renderPassageTranslations = jest.fn();
App.prototype.renderQuestions = jest.fn();
App.prototype.renderVocabularies = jest.fn();

describe('App component', () => {
  const props = {
    location: {
      pathname: 'test'
    },
    onInit: jest.fn()
  };
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<App {...props} />);
  });

  describe('about render', () => {
    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('component rendered', () => {
      expect(wrapper.find('.App').length).toBe(1);
    });

    it('Loader component rendered when isLoading is true', () => {
      wrapper.setProps({ isLoading: true });
      expect(wrapper.find(Loader).length).toBe(1);
    });

    describe('when packType is exist', () => {
      it('two ContentsBox components rendered', () => {
        wrapper.setProps({ packType: 'test' });
        expect(wrapper.find(ContentsBox).length).toBe(2);
      });

      it('two ContentsBox components rendered when isCheckComplete true', () => {
        wrapper.setProps({ isCheckComplete: true });
        expect(wrapper.find(ContentsBox).length).toBe(2);
      });
    });
  });

  describe('about state', () => {
    it('initial state set', () => {
      expect(wrapper.state()).toEqual({});
    });
  });

  describe('about props', () => {
    const wrapper = mount(<App {...props} />);

    it('allows to set props', () => {
      expect(wrapper.props()).toEqual(props);
    });
  });

  describe('about lifecycle method', () => {
    it('ComponentDidMout called once', () => {
      const props = {
        location: {
          pathname: 'test'
        },
        onInit: jest.fn()
      };
      const wrapper = mount(<App {...props} />);

      expect(props.onInit).toHaveBeenCalledTimes(1);
    });
  });
});
