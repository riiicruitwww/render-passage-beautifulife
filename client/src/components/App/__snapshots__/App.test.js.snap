// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    location={
      Object {
        "pathname": "test",
      }
    }
    onInit={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
          Array [
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="App__body"
        />,
        undefined,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isAnswerComplete": undefined,
          "isCheckComplete": undefined,
          "onCheckClick": undefined,
          "onHomeClick": undefined,
          "onNextClick": undefined,
          "packType": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "App__body",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="App__body"
          />,
          undefined,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAnswerComplete": undefined,
            "isCheckComplete": undefined,
            "onCheckClick": undefined,
            "onHomeClick": undefined,
            "onNextClick": undefined,
            "packType": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "App__body",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
